log4j-1.2-api/src/test/java/org/apache/log4j/CategoryTest.java

```diff
@@ -78,23 +78,23 @@ public void testForcedLog() {
         category.info("Hello, World");
         final List<LogEvent> list = appender.getEvents();
         int events = list.size();
-        assertTrue("Number of events should be 1, was " + events, events == 1);
+        assertThat(events).describedAs("Number of events should be 1, was " + events).isEqualTo(1);
         LogEvent event = list.get(0);
         Message msg = event.getMessage();
-        assertNotNull("No message", msg);
-        assertTrue("Incorrect Message type", msg instanceof ObjectMessage);
+        assertThat(msg).describedAs("No message").isNotNull();
+        assertThat(msg instanceof ObjectMessage).describedAs("Incorrect Message type").isTrue();
         Object[] objects = msg.getParameters();
-        assertTrue("Incorrect Object type", objects[0] instanceof String);
+        assertThat(objects[0] instanceof String).describedAs("Incorrect Object type").isTrue();
         appender.clear();
         category.log(Priority.INFO, "Hello, World");
         events = list.size();
-        assertTrue("Number of events should be 1, was " + events, events == 1);
+        assertThat(events).describedAs("Number of events should be 1, was " + events).isEqualTo(1);
         event = list.get(0);
         msg = event.getMessage();
-        assertNotNull("No message", msg);
-        assertTrue("Incorrect Message type", msg instanceof ObjectMessage);
+        assertThat(msg).describedAs("No message").isNotNull();
+        assertThat(msg instanceof ObjectMessage).describedAs("Incorrect Message type").isTrue();
```

Comment:
I do notice a pattern. Every one of the replacements is longer than the original. Of course, with IDE code completion that doesn't necessarily imply more typing.

Reply:
That’s true. In many cases we can reduce the number of assertions, for example then null check can be removed in favor of an assertThat(message).isInstanceOf(ObjectMessage.class) which provides a human readable failure in the null case as well.

the other downside is without an auto-formatter these lines aren’t broken properly. I’ve tried setting up a formatted but haven’t made it very far yet.

Reply:
I should have probably used “as” instead of “describedAs”, they do the exact same thing but the former is nine characters shorter.